aten/src/ATen/test/basic.cpp

```diff
@@ -167,121 +165,141 @@ static void test(Type & type) {
   }
-  CATCH_SECTION("indexing by Scalar") {
-    Tensor tensor = arange(0, 10, kInt);
-    Tensor one = ones({}, kInt);
-    for (int64_t i = 0; i < tensor.numel(); ++i) {
-      CATCH_REQUIRE(tensor[i].equal(one * i));
-    }
-    for (size_t i = 0; i < static_cast<uint64_t>(tensor.numel()); ++i) {
-      CATCH_REQUIRE(tensor[i].equal(one * static_cast<int64_t>(i)));
-    }
-    for (int i = 0; i < tensor.numel(); ++i) {
-      CATCH_REQUIRE(tensor[i].equal(one * i));
-    }
-    for (int16_t i = 0; i < tensor.numel(); ++i) {
-      CATCH_REQUIRE(tensor[i].equal(one * i));
-    }
-    for (int8_t i = 0; i < tensor.numel(); ++i) {
-      CATCH_REQUIRE(tensor[i].equal(one * i));
-    }
-    CATCH_REQUIRE_THROWS_WITH(
-        tensor[Scalar(3.14)].equal(one),
-        StartsWith(
-            "Can only index tensors with integral scalars"));
+  for (int i = 0; i < tensor.numel(); ++i) {
+    EXPECT_TRUE(tensor[i].equal(one * i));
   }
-  CATCH_SECTION("indexing by zero-dim tensor") {
-    Tensor tensor = arange(0, 10, kInt);
-    Tensor one = ones({}, kInt);
-    for (int i = 0; i < tensor.numel(); ++i) {
-      CATCH_REQUIRE(tensor[one * i].equal(one * i));
-    }
-    CATCH_REQUIRE_THROWS_WITH(
-        tensor[ones({}) * 3.14].equal(one),
-        StartsWith(
-            "Can only index tensors with integral scalars"));
-    CATCH_REQUIRE_THROWS_WITH(
-        tensor[Tensor()].equal(one),
-        StartsWith("Can only index with tensors that are defined"));
-    CATCH_REQUIRE_THROWS_WITH(
-        tensor[ones({2, 3, 4}, kInt)].equal(one),
-        StartsWith("Can only index with tensors that are scalars (zero-dim)"));
+  for (int16_t i = 0; i < tensor.numel(); ++i) {
+    EXPECT_TRUE(tensor[i].equal(one * i));
   }
-  CATCH_SECTION("dispatch") {
-    Tensor tensor = randn({20, 20});
-    Tensor other = randn({20, 20});
-    auto result = tensor.m(relu).m(mse_loss, other, Reduction::ElementwiseMean);
-    CATCH_REQUIRE(result.allclose(mse_loss(relu(tensor), other)));
+  for (int8_t i = 0; i < tensor.numel(); ++i) {
+    EXPECT_TRUE(tensor[i].equal(one * i));
   }
-  CATCH_SECTION("core") {
-    int i = CoreTest();
-    CATCH_REQUIRE(i + 1 == CoreTest());
+  EXPECT_ANY_THROW(tensor[Scalar(3.14)].equal(one));
+}
+
+void test_indexing_by_zerodim_tensor() {
+  Tensor tensor = arange(0, 10, kInt);
+  Tensor one = ones({}, kInt);
+  for (int i = 0; i < tensor.numel(); ++i) {
+    EXPECT_TRUE(tensor[one * i].equal(one * i));
   }
+  EXPECT_ANY_THROW(tensor[ones({}) * 3.14].equal(one));
+  EXPECT_ANY_THROW(tensor[Tensor()].equal(one));
+  EXPECT_ANY_THROW(tensor[ones({2, 3, 4}, kInt)].equal(one));
+}
+void test_dispatch() {
+  Tensor tensor = randn({20, 20});
+  Tensor other = randn({20, 20});
+  auto result = tensor.m(relu).m(mse_loss, other, Reduction::ElementwiseMean);
+  EXPECT_TRUE(result.allclose(mse_loss(relu(tensor), other)));
+}
+
+void test_core() {
+  int i = CoreTest();
+  EXPECT_EQ(i + 1, CoreTest());
+}
+
+void test(Type& type) {
```

Comment:
Generally this is not the recommended way of running a collection of tests with different parameters in gtest. If you want to be very thorough, we should rewrite these tests using "Parameterized tests": https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests.

It may be more work though. Up to you